src/formats/internal/reader.c

```diff
@@ -15,12 +17,40 @@ struct sd_reader {
 };
 
 sd_reader *sd_reader_open(const char *file) {
-    sd_reader *reader = omf_calloc(1, sizeof(sd_reader));
+    char *path_dup = omf_strdup(file);
+    char *path_dup2 = omf_strdup(file);
+    char *fn = omf_basename(path_dup);
+    char *directory = omf_dirname(path_dup2);
+
+    list dir_list;
+    list_create(&dir_list);
+    if(scan_directory(&dir_list, directory) != 0) {
```

Comment:
What if we used scan_directory only when we were looking for a file with a potentially screwy cased name?

Reply:
or if we had like a `scan_directory_for_file` function we could call?

Reply:
Yeah, could make the scan optional.

Reply:
scan_directory_for_file was added.